circuits/src/field/decomposition/pow2range.rs

```diff
@@ -237,6 +237,52 @@ impl<F: PrimeField> Pow2RangeChip<F> {
     }
 }
 
+#[cfg(feature = "extraction")]
+pub mod extraction {
+    //! Extraction specific logic related to the p2r chip.
+
+    use extractor_support::circuit::{ChipArgs, CircuitInitialization};
+    use ff::PrimeField;
+    use midnight_proofs::{
+        circuit::Layouter,
+        plonk::{Advice, Column, ConstraintSystem, Error},
+    };
+
+    use super::Pow2RangeChip;
+    use crate::field::decomposition::pow2range::Pow2RangeConfig;
+
+    impl<F: PrimeField, L: Layouter<F>> CircuitInitialization<L> for Pow2RangeChip<F> {
+        type Config = Pow2RangeConfig;
+
+        type Args = usize;
+
+        type ConfigCols = [Column<Advice>; 4];
+
+        type CS = ConstraintSystem<F>;
+
+        type Error = Error;
+
+        fn new_chip(config: &Self::Config, args: Self::Args) -> Self {
+            Self::new(config, args)
+        }
+
+        fn configure_circuit(
+            meta: &mut ConstraintSystem<F>,
+            columns: &Self::ConfigCols,
+        ) -> Self::Config {
+            Self::configure(meta, columns)
+        }
+
+        fn load_chip(&self, layouter: &mut L, _config: &Self::Config) -> Result<(), Error> {
+            self.load_table(layouter)
+        }
+    }
+
+    impl<F: PrimeField> ChipArgs for Pow2RangeChip<F> {
+        type Args = usize;
+    }
```

Comment:
For enabling extraction for a chip these two traits need to be implemented. The general structure mimics `FromScratch` and `ComposableChip`. Consequently, most implementations of these traits can be derived from an existing implementation of `FromScratch` and I included two derive macros that automatically implements these traits if the chip implements `FromScratch`.